packages/zoe/src/contracts/scaledPriceAuthority.js

```diff
@@ -52,7 +52,8 @@ export const prepare = async (zcf, privateArgs, baggage) => {
 
   const priceAuthority = makePriceAuthorityTransform({
     quoteMint,
-    sourcePriceAuthority,
+    // If the priceAuthority is overridden in privateArgs, use that version
+    sourcePriceAuthority: privateArgs?.priceAuthority || sourcePriceAuthority,
```

Comment:
since both terms and privateArgs can pass this in, please make the names more distinct. terms can't be changed so something like `newPriceAuthority` for privateArgs